src/h/opengl.h

```diff
@@ -12,9 +33,1215 @@
 #define U3D_QUAD_STRIP     GL_QUAD_STRIP
 #define U3D_POLYGON        GL_POLYGON
 
+#define U2D_LINE_LOOP      GL_LINE_LOOP
+#define U2D_LINE_STRIP	   GL_LINE_STRIP
+#define U2D_POLYGON	   GL_POLYGON
+
 /*
  * texture modes.
  */
 #define U3D_REPLACE        GL_REPLACE
 #define U3D_BLEND          GL_BLEND
 #define U3D_MODULATE       GL_MODULATE
+
+/*
+ * Graphical primitive IDs
+ */
+#define GL2D_BLIMAGE		9000
+#define GL2D_FILLPOLYGON	9001
+#define GL2D_DRAWPOLYGON	9002
+#define GL2D_DRAWSEGMENT	9003
+#define GL2D_DRAWLINE		9004 
+#define GL2D_DRAWPOINT		9005
+#define GL2D_DRAWCIRCLE		9006
+#define GL2D_FILLCIRCLE		9007
+#define GL2D_DRAWARC		9008
+#define GL2D_FILLARC		9009
+#define GL2D_DRAWRECTANGLE	9010
+#define GL2D_FILLRECTANGLE	9011
+#define GL2D_COPYAREA		9012
+#define GL2D_ERASEAREA		9013
+#define GL2D_STRIMAGE		9014
+#define GL2D_READIMAGE		9015
+#define GL2D_DRAWSTRING		9016
+#define GL2D_WWRITE		9017
+
+/* 
+ * Attribute primitive IDs
+ */
+/* colors */
+#define GL2D_FG			9050
+#define GL2D_BG			9051
+#define GL2D_REVERSE		9052
+#define GL2D_GAMMA		9053
+#define GL2D_DRAWOP		9054
+
+/* fonts */
+#define GL2D_FONT		9055
+#define GL2D_LEADING		9056
+
+/* lines/fills */
+#define GL2D_LINEWIDTH		9057
+#define GL2D_LINESTYLE		9058
+#define GL2D_FILLSTYLE		9059
+#define GL2D_PATTERN		9060
+
+/* positioning */
+#define GL2D_CLIP		9061
+#define GL2D_DX			9062
+#define GL2D_DY			9063
+
+/*
+ * line styles
+ */
+#define GL2D_LINE_SOLID		0
+#define GL2D_LINE_DASHED	1
+#define GL2D_LINE_STRIPED	2
+
+/*
+ * fill styles
+ */
+#define GL2D_FILL_SOLID		0
+#define GL2D_FILL_TEXTURED	1
+#define GL2D_FILL_MASKED	2
```

Comment:
All of the define lines  from line 50 down to here could use some format/align.

Reply:
I took a look at my source code, and it appears that the misalignments are due to tabs being pushed by the prepended '+'